src/optimizer/remove_unused_columns.cpp

```diff
@@ -176,10 +177,22 @@ void RemoveUnusedColumns::VisitOperator(LogicalOperator &op) {
 					order.projections.push_back(col_idx);
 				}
 			}
-		}
-		for (auto &child : op.children) {
-			RemoveUnusedColumns remove(binder, context, true);
-			remove.VisitOperator(*child);
+
+			// This seems to break when we have columns that are constant expressions like '1' or true.
+			// I'm not sure what the solution is.
+			// Maybe add all_bindings here (somehow).
+			// Or, need to update lower code so that unused expressions can be omitted.
+			for (auto &child : op.children) {
+				RemoveUnusedColumns remove(binder, context);
+				remove.VisitOperatorExpressions(op);
+				remove.column_references.insert(column_references.begin(), column_references.end()); // add parent references
+				remove.VisitOperator(*child);
+			}
+		} else {
+			for (auto &child : op.children) {
+				RemoveUnusedColumns remove(binder, context, true);
+				remove.VisitOperator(*child);
+			}
 		}
```

Comment:
The above code change for ORDER BY projection pushdown is where I could use some help. It breaks down for constant columns. I don't fully understand the optimizer code and could use some help in how to handle this case. For example:
```
# name: test/issues/rigger/test_10087.test
# description: SQLancer bug that detected an issue with the column lifetime analyzer
# group: [rigger]

statement ok
PRAGMA enable_verification

statement ok
CREATE TABLE t0(c1 INT);

statement ok
INSERT INTO t0(c1) VALUES (1);

statement ok
CREATE VIEW v0(c0, c1, c2) AS SELECT '1', true, t0.c1 FROM t0 ORDER BY -1-2 LIMIT 2;

statement ok
SELECT v0.c2 FROM v0 WHERE (NOT (v0.c1 IS NOT NULL));
```

This gives the error:
```
INTERNAL Error: Failed to bind column reference "" [1.0] (bindings: {#[1.1], #[1.2]})
```
Basically, the optimizer is trying to skip the expressions `1` and true as unused by the higher level select but the engine still seems to want to execute them.

Reply:
In the general case, this code seems to work and correctly applies projection pushdown. So, it is this edge case that I am hoping to resolve.